File: backend/app/models/agent.py
"""Agent model."""

from datetime import datetime, timezone

from app import db


class Agent(db.Model):
    """Agent model representing an AI agent in the system."""

    __tablename__ = "agents"

    id = db.Column(db.Integer, primary_key=True)
    name = db.Column(db.String(100), nullable=False)
    type = db.Column(
        db.String(50), nullable=False
    )  # 'driver', 'creator', 'generator', 'dynamic'
    role = db.Column(db.String(200), nullable=False)
    status = db.Column(db.String(50), default="idle")  # 'idle', 'working', 'waiting'
    config = db.Column(db.JSON, nullable=True)
    created_at = db.Column(db.DateTime, default=lambda: datetime.now(timezone.utc))

    # Relationships
    messages = db.relationship("Message", backref="agent", lazy=True)
    tasks = db.relationship("Task", backref="assigned_agent", lazy=True)

    def to_dict(self) -> dict:
        """Convert agent to dictionary."""
        return {
            "id": self.id,
            "name": self.name,
            "type": self.type,
            "role": self.role,
            "status": self.status,
            "config": self.config,
            "created_at": self.created_at.isoformat() if self.created_at else None,
        }

File: frontend/src/types/agent.ts
export type AgentType = "driver" | "creator" | "generator" | "dynamic";
export type AgentStatus = "idle" | "working" | "waiting";
export type MessageSender = "agent" | "operator" | "system";

export interface Agent {
  id: number;
  name: string;
  type: AgentType;
  role: string;
  status: AgentStatus;
  config?: Record<string, unknown>;
  created_at: string;
  updated_at?: string;
}

export interface Message {
  id: number;
  agent_id: number;
  sender: MessageSender;
  content: string;
  timestamp: string;
  metadata?: Record<string, unknown>;
}

